### DISCUSSION TITLE
v6-alpha. Layer origin co-ordinate issue

### DISCUSSION THREAD

**Discussion by mattevans-dev** [fabricjs/fabric.js#7958] [Q&A] [answered]
## Version
6.0.0

## Test Case
- [ ] [Browser](https://codepen.io/grumpygrouse/pen/BaYZZVV) (https://codepen.io/grumpygrouse/pen/BaYZZVV)

## Information about environment
Chrome

## Steps To Reproduce

1. Click Animate

## Expected Behavior
Expect layers to have the same origin as the canvas, e.g top left.

## Actual Behavior
When adding objects to a layer, they are correctly positioned relative to top/left within a layer. However, if you were to animate or modify an object within the layer, the top/left position is subsequently relative to the layers centre, rather than top/left of the canvas. E.g in our example, the initial left is 50. However, when you click animate and log out the value, it shows -50.

**Comment by ShaMan123** [comment]
Absolutely correct. This is how groups are designed. 
I have a commit somewhere that changes the origin as you mentioned but I decided not to move forward with it (don't remember why... I think because I made Layer compatible as a Group descendant so I wanted to keep everything uniform)
However I did expose `getX/setX` which eliminates the need to worry about coordinate systems because it relates to the canvas no matter how deeply nested an object is.

**Comment by ShaMan123** [comment]
Take a look https://github.com/fabricjs/fabric.js/blob/fbb4581722b14950e0679ccbedec7ba3714ac63c/src/mixins/animation.mixin.js#L26

**Comment by ShaMan123** [comment]
I think this can be closed. reopen/comment if you think otherwise.

**Comment by mattevans-dev** [comment]
Thanks @ShaMan123  - we're looking at animating elements inside a layer and the layer itself at the same time. So the getX/setX doesn't quite work for us. You mentioned you changed the origin. How did you go about this, changing the coordinate system on the group? Could you point me in the right direction please?

**Reply by ShaMan123** [reply]
You can use `fabric.util.sendPointToPlane` - very useful as well for such cases

**Comment by ShaMan123** [comment, marked as answer]
Found it 9a3e2b1f
Experimental stuff.

This experiment makes layer a dummy container simply for caching and **NOTHTING** else

**Comment by ShaMan123** [comment]
BTW I would like to understand why the get/setXY methods don't do the job.
And would appreciate some visuals if you are comfortable sharing.

**Reply by ShaMan123** [reply]
And another thought - Layer is meant to be static - maybe you want a group with custom layout?

**Comment by mattevans-dev** [comment]
HI @ShaMan123,
Thanks once again for the help, and for your brilliant contributions to the new group system. I'm building a video editor which consists of 'scenes'. The user can edit elements at an individual scene level, but the final video is multiple scenes stitched together with a variety of transitional effects. To keep things simple, I want the coordinate system to be the same when editing at a scene level, and when playing back at a project level. Your example came in handy as I'm now shifting the offscreen canvas before it draws allowing me to achieve the result I'm after.

**Reply by ShaMan123** [reply]
Sounds cool.
Ping me if you have any ideas/bugs regarding the new Group and it's derivatives.
